src/js/views/Project/Create/Create.jsx

```diff
@@ -1,156 +1,227 @@
-import PropTypes from 'prop-types'
-import React, { useContext, useEffect, useReducer, useState } from 'react'
+import PropTypes, { string } from 'prop-types'
+import React, { useContext, useEffect, useState } from 'react'
 import { useNavigate } from 'react-router-dom'
 import { useTranslation } from 'react-i18next'
+import { validate } from 'jsonschema'
 
 import { Context } from '../../../state'
 import { ErrorBoundary, Form } from '../../../components'
-import { User } from '../../../schema'
 
-import { Attributes } from './Attributes'
-import { Automations } from './Automations'
-import { initialState, reducer } from './Reducer'
-import { GitlabConnectModal } from './GitlabConnectModal'
-import { SavingDialog } from './SavingDialog'
-import { Links } from './Links'
-import { URLs } from './URLs'
+import { jsonSchema } from '../../../schema/Project'
+import { metadataAsOptions } from '../../../settings'
+import { httpGet, httpPost } from '../../../utils'
 
-function Create({ user }) {
-  const { t } = useTranslation()
+import { AutomationList } from './AutomationList'
 
-  const attributesLink = { href: '#attributes', label: t('project.attributes') }
-  const automationsLink = {
-    href: '#automations',
-    label: t('project.automations')
+const EMPTY_FIELDS = {
+  namespace_id: null,
+  project_type_id: null,
+  name: null,
+  slug: null,
+  description: null
+}
+
+function Automation({ automationName, integrationName, automationSlug }) {
+  return {
+    integrationName,
+    automationName,
+    automationSlug
   }
+}
```

Comment:
I'm not really sure what the point of this function is since it just returns a copy of what it got as input. Is it so you can have the proptypes defined below?

Reply:
I think that was it ... I really wanted to [use it in AutomationList.jsx](https://github.com/AWeber-Imbi/imbi-ui/blob/99c811f251260037c013d36e5bfb2015da4c64a0/src/js/views/Project/Create/AutomationList.jsx#L50) but I never figured out how to make it work nicely. Lemme clean that half-step up as well.

Reply:
I think you're looking for

```javascript
Automation = PropTypes.exact({
  automationName: PropTypes.string.isRequired,
  automationSlug: PropTypes.string.isRequired,
  integrationName: PropTypes.string.isRequired
})
...
AutomationList.propTypes = {
  automations: PropTypes.arrayOf(Automation).isRequired,
  selectedAutomations: PropTypes.arrayOf(string).isRequired,
  setSelectedAutomations: PropTypes.func.isRequired
}
```

Reply:
I ended up dropping this altogether since I couldn't come up with a good way to get WebStorm to pay attention to the requirement so at best it results in a runtime explosion in the console and empty toggles on the screen 😞